src/test/java/com/epam/test_generator/services/ProjectServiceTest.java

```diff
@@ -0,0 +1,398 @@
+        when(authentication.getPrincipal()).thenReturn(new AuthenticatedUser(
+            null,
+            simpleUser1.getEmail(),
+            null,
+            null,
+            false
+        ));
+        when(projectDAO.findOne(simpleProject1.getId())).thenReturn(simpleProject1);
+        when(userService.getUserByEmail(simpleUser1.getEmail())).thenReturn(simpleUser1);
+        when(projectService.getProjectByProjectId(simpleProject1.getId())).thenReturn(simpleProject1);
+        ProjectFullDTO actualFullProject = projectService
+            .getAuthUserFullProject(simpleProjectFullDTO1.getId(), authentication);
+    }
+
+    @Test
+    public void createProject() {
+    }
+
+    @Test
+    public void updateProject() {
+    }
+
+    @Test
+    public void removeProject_valid_Success() {
+        when(projectDAO.findOne(SIMPLE_PROJECT_ID)).thenReturn(simpleProject1);
+        projectService.removeProject(SIMPLE_PROJECT_ID);
+        verify(projectDAO).delete(SIMPLE_PROJECT_ID);
+    }
+
+    @Test(expected = NotFoundException.class)
+    public void removeProject_invalidProjectId_NotFound() {
+
+        when(projectDAO.findOne(SIMPLE_PROJECT_ID)).thenReturn(simpleProject1);
+        projectService.removeProject(777L);
+        verify(projectDAO).delete(SIMPLE_PROJECT_ID);
+    }
+
+    @Test
+    public void addUserToProject_valid_Success() {
+        simpleProject1.setUsers(new HashSet<>());
+        when(projectDAO.findOne(SIMPLE_PROJECT_ID)).thenReturn(simpleProject1);
+        when(userService.getUserById(SIMPLE_USER_ID)).thenReturn(simpleUser1);
+
+        assertFalse(simpleProject1.getUsers().contains(simpleUser1));
+        projectService.addUserToProject(SIMPLE_PROJECT_ID, SIMPLE_USER_ID);
+        assertTrue(simpleProject1.getUsers().contains(simpleUser1));
+        verify(projectDAO).save(simpleProject1);
+    }
+
+    @Test(expected = NotFoundException.class)
+    public void addUserToProject_invalidProjectId_NotFound() {
+        when(projectDAO.findOne(SIMPLE_PROJECT_ID)).thenReturn(simpleProject1);
+        when(userService.getUserById(SIMPLE_USER_ID)).thenReturn(simpleUser1);
+
+        projectService.addUserToProject(777L, SIMPLE_USER_ID);
+    }
+
+    @Test(expected = NotFoundException.class)
+    public void addUserToProject_invalidUserId_NotFound() {
+        when(projectDAO.findOne(SIMPLE_PROJECT_ID)).thenReturn(simpleProject1);
+        when(userService.getUserById(SIMPLE_USER_ID)).thenReturn(simpleUser1);
+
+        projectService.addUserToProject(SIMPLE_PROJECT_ID, 666L);
+    }
+
+    @Test
+    public void removeUserFromProject_valid_Success() {
+        simpleProject1.setUsers(Sets.newHashSet(simpleUser1));
+        when(projectDAO.findOne(SIMPLE_PROJECT_ID)).thenReturn(simpleProject1);
+        when(userService.getUserById(SIMPLE_USER_ID)).thenReturn(simpleUser1);
+
+        assertTrue(simpleProject1.getUsers().contains(simpleUser1));
```

Comment:
This assertion does not test production code

Reply:
Why is it not? we mock projectDAO#findOne and userService#getUserById methods and projectDAO#save - they do all the stuff; this method just checks all data is ok and then add user to list of Projects. Basically we checks that simple project now contains simple user (but before execution of #addUserToProject) it was not